### PR TITLE
Move custom type renaming to custom types table

### PR DESCRIPTION
**Pull request by danigo99**
## Context

Ticket: https://linear.app/prismic/issue/SM-858/[delete-ct]-aauser-i-can-rename-a-custom-type-from-the-ct-list

## The Solution

Removed the edit button in the custom type editor, added onClick functionality to the rename option of the kebab menu.

## Impact / Dependencies

Waiting on #667 

## Checklist before requesting a review
- [ ] I hereby declare my code ready for review.
- [x] If it is a critical feature, I have added tests.
- [x] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

## [OPT] Preview

ezgif com-gif-maker (4)

---

image

### CODE REVIEW

File: packages/slice-machine/components/CustomTypeTable/ctPage.tsx [hunk 1]
```diff
@@ -1,98 +1,120 @@
...
 import { KebabMenuDropdown } from "@components/KebabMenuDropdown";
+import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+import { RenameCustomTypeModal } from "@components/Forms/RenameCustomTypeModal";
 
 export const CustomTypeTable: React.FC<{
   customTypes: FrontEndCustomType[];
 }> = ({ customTypes }) => {
   const { modelsStatuses, authStatus, isOnline } = useModelStatus(customTypes);
+  const [renameCustomTypeName, setRenameCustomTypeName] = useState<string>("");
+  const [renameCustomTypeId, setRenameCustomTypeId] = useState<string>("");
```
Comment (danigo99) [inline packages/slice-machine/components/CustomTypeTable/ctPage.tsx]: I'm not super happy with this way of doing things. But other than adding more complexity to the global state (in order to pass these through) I'm not sure how else to do it 😕

Comment (Heyrows) [reply packages/slice-machine/components/CustomTypeTable/ctPage.tsx]: Maybe the Rename modal should have it's own state.

What I mean is, clicking the rename button would just open the modal with the right Slice / Custom type.
The modal contains it's inputs & handle it's state by itself.
On clicking the save button it triggers a saga to rename and at the end of the saga we close the modal.

What do you think ?

File: packages/slice-machine/src/modules/selectedCustomType/reducer.ts [hunk 2]
```diff
@@ -317,22 +316,6 @@ export const selectedCustomTypeReducer: Reducer<
         )((group: GroupSM) => Group.reorderWidget(group, start, end))
       );
     }
-    case getType(renameCustomTypeCreator.success): {
```
Comment (Heyrows) [inline packages/slice-machine/src/modules/selectedCustomType/reducer.ts:320]: nice, as a change that doesn't need to be saved it makes way more sense to have it here than in the builder.

File: packages/slice-machine/components/Forms/RenameCustomTypeModal/RenameCustomTypeModal.tsx [hunk 3]
```diff
@@ -39,6 +41,13 @@ export const RenameCustomTypeModal: React.FC<RenameCustomTypeModalProps> = ({
     isRenamingCustomType: isLoading(store, LoadingKeysEnum.RENAME_CUSTOM_TYPE),
   }));
 
+  const customTypeName = useMemo(() => {
```
Comment (Heyrows) [inline packages/slice-machine/components/Forms/RenameCustomTypeModal/RenameCustomTypeModal.tsx]: we could even have passed just a custom type found previously.

**Review by Heyrows — APPROVED** [review]

File: packages/slice-machine/src/modules/availableCustomTypes/index.ts [hunk 4]
```diff
@@ -56,7 +56,8 @@ type CustomTypesActions =
   | ActionType<typeof createCustomTypeCreator>
   | ActionType<typeof renameCustomTypeCreator>
   | ActionType<typeof saveCustomTypeCreator>
-  | ActionType<typeof pushCustomTypeCreator>;
+  | ActionType<typeof pushCustomTypeCreator>
+  | ActionType<typeof renameCustomTypeCreator>;
```
Comment (mathieud) [inline packages/slice-machine/src/modules/availableCustomTypes/index.ts]: This seems duplicated

**Review by mathieud — APPROVED** [review]

**Pull request merged**